### PR TITLE
Feat: CRUD

### PR DESCRIPTION
**Pull request by Setanna**

### CODE REVIEW

**Comment by Setanna** [comment]
[API - CRUD](https://trello.com/c/eto2GV4J/2-api-crud)

File: app/Http/Controllers/CategoryController.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    /**
+     * Update the specified resource in storage.
+     */
+    public function update(Request $request, Category $category)
+    {
+        $category->update($request->all());
+
+        return new CategoryResource($category);
+    }
```
Comment (MathiasDuus) [inline app/Http/Controllers/CategoryController.php]: Here we blindly trust the input

Comment (MathiasDuus) [reply app/Http/Controllers/CategoryController.php]: Also check all of the other update functions

File: app/Http/Controllers/ProductController.php [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+        $products = Product::with(['location_products' => function($query) use ($location){
+            return $query->where('location_id', $location->id);
+        }])->get();
+
+        /*
+        $products = Product::whereHas('location_products', function($query) use ($location) {
+            $query->where('location_id', $location->id)->first();
+        })->get();
+        */
```
Comment (MathiasDuus) [inline app/Http/Controllers/ProductController.php]: Should this be removed?

**Review by MathiasDuus — COMMENTED** [review]
We can become even more type safe by adding the return types to the functions and adding the parameters and returns to the docblock of the function

File: app/Http/Controllers/CategoryController.php (hunk 1 above)
Comment (Setanna) [reply app/Http/Controllers/CategoryController.php]: The two controllers (Category & Location) with incorrect input validation have been fixed

File: app/Http/Controllers/ProductController.php (hunk 2 above)
Comment (Setanna) [reply app/Http/Controllers/ProductController.php]: Unused code has been removed

**Review by MathiasDuus — APPROVED** [review]

**Pull request merged**